solidity/contracts/converter/types/liquid-token/LiquidTokenConverter.sol

```diff
@@ -74,10 +75,10 @@ contract LiquidTokenConverter is ConverterBase {
       * @return expected target amount
       * @return expected fee
     */
-    function targetAmountAndFee(IERC20Token _sourceToken, IERC20Token _targetToken, uint256 _amount) public view returns (uint256, uint256) {
-        if (_targetToken == ISmartToken(anchor) && reserves[_sourceToken].isSet)
+    function targetAmountAndFee(IERC20Token _sourceToken, IERC20Token _targetToken, uint256 _amount) public override view returns (uint256, uint256) {
+        if (_targetToken == ISmartToken(address(anchor)) && reserves[_sourceToken].isSet)

```

Comment:
Why two casts are needed here - `ISmartToken(address(anchor))`?

Reply:
you can cast from a contract to one of its ancestors but not the other way around.